### PR TITLE
Assays parsing

### PR DESCRIPTION
**Pull request by a-solovyev12**
This PR adds functionality to parse an SCE objects with 'counts' slot.
This is needed because DropletUtils has 'counts' hardcoded as an assay. 
`scpred_train_test_split.R` checks for assay type in SCE object and converts it into 'normcounts' if necessary.

### CODE REVIEW

File: scpred_train_test_split.R [hunk 1]
```diff
@@ -84,6 +84,10 @@ set.seed(opt$random_seed)
 print(opt$input_sce_object)
 # preprocess data from SCE object 
 sce_object = readRDS(opt$input_sce_object)
+# check for 'normcounts' slot
```
Comment (pinin4fjords) [inline scpred_train_test_split.R]: This sort of silent correction always makes me nervous. An object having 'normcounts' has meaning- i.e. it means that the counts have been normalised. A user could pass their un-normalised counts in by mistake. The better thing to do is to add an 'assay' parameter to the script with help text like:

"name of assay in the SCE object containing normalised counts, 'normcounts' by default". 

That way, the default behaviour of reminding the user to normalise persists, but we can override if we know the counts we're passing are normalised.

Of course we could also fix our DropletUtils wrapper to have a parameter about the assay name to use.

Comment (a-solovyev12) [reply scpred_train_test_split.R]: I think then we'll need to modify the DropletUtils package (I can make a fork). The reason is that when the script `read10xCounts.R` initialises an SCE object, it puts 'counts' into the assays slot by default, even if the data you provided is normalised. So, we need to have control on that level.

Comment (a-solovyev12) [reply scpred_train_test_split.R]: To be fair, I think just reminding the user would suffice.

File: scpred_train_test_split.R [hunk 2]
```diff
@@ -84,7 +90,8 @@ set.seed(opt$random_seed)
 print(opt$input_sce_object)
 # preprocess data from SCE object 
 sce_object = readRDS(opt$input_sce_object)
-sce_counts = as.matrix(normcounts(sce_object))
+sce_counts = as.matrix(assays(sce_object)[[opt$normalised_counts_slot]])
```
Comment (pinin4fjords) [inline scpred_train_test_split.R]: Can you add a check that the slot exists, and exit with an error if it doesn't?

Comment (a-solovyev12) [reply scpred_train_test_split.R]: @pinin4fjords done

**Review by pinin4fjords — APPROVED** [review]
Looks good
